fix: checking value of send_welcome_email

ABHISHEK-KEDAR-21 opened the pull request:
If send_welcome_email is set to 0 still it is triggering send_welcome_mail_to_user(). Need to check if send_welcome_email=1.
This is required.

<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/frappe/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

<!-- You can skip this if you're fixing a typo or updating existing documentation -->

> Explain the **details** for making this change. What existing problem does the pull request solve?

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

ABHISHEK-KEDAR-21 committed:
checking value of send_welcome_email

If send_welcome_email is set to 0 still it is triggering send_welcome_mail_to_user(). Need to check if send_welcome_email=1.
This is required.

Thunderbottom commented on frappe/core/doctype/user/user.py:
```diff
@@ -205,7 +205,7 @@ def send_password_notification(self, new_password):
 							logout_all_sessions=self.logout_all_sessions)
 
-					if not self.flags.no_welcome_mail and self.send_welcome_email:
+					if not self.flags.no_welcome_mail and self.send_welcome_email==1:
```
This does not make sense. The `== 1` in here is redundant, and it should look for if the value is set to 1 (or True) by default. The issue is that the value comes in a string, which you may have to set to int. I'd suggest trying this out instead:

```suggestion
					if not self.flags.no_welcome_mail and cint(self.send_welcome_email):
```

ABHISHEK-KEDAR-21 committed:
fix: cint(self.send_welcome_email) in user.py

Instead of checking self.send_welcome_email==1 used cint(self.send_welcome_email

Co-Authored-By: Chinmay Pai <chinmaydpai@gmail.com>

ABHISHEK-KEDAR-21 commented on frappe/core/doctype/user/user.py:
```diff
@@ -205,7 +205,7 @@ def send_password_notification(self, new_password):
 							logout_all_sessions=self.logout_all_sessions)
 
-					if not self.flags.no_welcome_mail and self.send_welcome_email==1:
+					if not self.flags.no_welcome_mail and cint(self.send_welcome_email):
```
@Thunderbottom Thanks for the suggestion. I was not aware of the cint(), will definitely use it in my coding.

Thunderbottom committed:
chore: remove list from kwargs

Thunderbottom commented:
could you please raise the same pull request on `version-12-hotfix` and `version-11-hotfix` as well?

ABHISHEK-KEDAR-21 commented:
Yes @Thunderbottom 

> could you please raise the same pull request on `version-12-hotfix` and `version-11-hotfix` as well?

AKedar21 commented:
> could you please raise the same pull request on `version-12-hotfix` and `version-11-hotfix` as well?

@Thunderbottom Yes
